src/cache/cache_operations.rs

```diff
@@ -20,12 +25,64 @@ pub struct Invalidate {
     database: Database,
 }
 
+#[derive(Deserialize, Debug)]
+pub enum InvalidationScope {
+    Requests,
+    Responses,
+    Both,
+}
+
+#[derive(Deserialize, Debug)]
+#[serde(tag = "invalidate", content = "pattern_details")]
 pub enum InvalidationPattern {
-    FlushAll,
-    RequestsResponses(String),
+    Any(String, InvalidationScope),
```

Comment:
Minor: I find the order of pattern, scope confusing. I would have the scope first and then the pattern

Reply:
That would change the JSON I placed in the description of this issue. I don't mind either though :) just FYI. I will update accordingly